lib/paquito/safe_yaml.rb

```diff
@@ -43,31 +43,38 @@ def dump(obj)
 
     class RestrictedYAMLTree < Psych::Visitors::YAMLTree
       class DispatchCache
-        def initialize(visitor, cache, permitted_classes)
+        def initialize(visitor, cache)
           @visitor = visitor
           @cache = cache
-          @permitted_classes = permitted_classes
-
-          @permitted_cache = Hash.new do |h, klass|
-            unless @permitted_classes.include?(klass.name)
-              raise UnsupportedType, "Tried to dump unspecified class: #{klass.name.inspect}"
-            end
-
-            h[klass] = true
-          end.compare_by_identity
         end
 
         def [](klass)
-          if @permitted_cache[klass]
-            @cache[klass]
-          end
+          return unless @visitor.permitted_class?(klass)
+
+          @cache[klass]
```

Comment:
I preferred the non guard clause version. Guard clauses are nice when there's an actual logic later, but here there's only one condition the guard clause itself.

Eventually I could settle for:

```ruby
@cache[klass] if @visitor.permitted_class?(klass)
```

Reply:
One-liner looks much better, not sure why I went with a guard clause. I agree that guard doesn't look suitable here. 
Changed!